bin/guard

```diff
@@ -2,31 +2,114 @@
 
 require "pathname"
 
-default = Pathname(ENV.fetch("BUNDLE_GEMFILE", "Gemfile"))
-relative_to_binstub = Pathname(__FILE__).realpath + "../../Gemfile"
-gemfile = [default, relative_to_binstub].detect(&:exist?)
-
-if gemfile && !ENV["RUBYGEMS_GEMDEPS"]
-  ENV["BUNDLE_GEMFILE"] = gemfile.to_s
-  require "rubygems"
-  require "bundler/setup"
-else
-  require "rubygems"
-end
+class GuardReloader
+  class Config
+    def using_rubygems?
+      ENV["RUBYGEMS_GEMDEPS"]
```

Comment:
`RUBYGEMS_GEMDEPS` will be used if you use rubygems starting with version 2. And `rubygems` is either required or not, there is no separate full rubygems and builtin. When you use a bin created by installing a gem by default it will be required. Please check the bin generated for guard (`cat $(which guard)` unless you use rbenv which creates an additional wrapper).